### PR TITLE
dialects: (riscv_scf) add custom syntax to riscv_scf.while

### PR DESCRIPTION
**Pull request by superlopuh**
Follows the syntax of scf.while exactly

### CODE REVIEW

**Review by adutilleul — APPROVED** [review]

File: xdsl/dialects/riscv_scf.py [hunk 1]
```diff
@@ -240,6 +240,75 @@ def verify_(self):
...
+            parser.Delimiter.PAREN,
+            parse_assignment,
+        )
+
+        parser.parse_punctuation(":")
+        type_pos = parser.pos
+        function_type = parser.parse_function_type()
+
+        if len(tuples) != len(function_type.inputs.data):
+            parser.raise_error(
```
Comment (superlopuh) [inline xdsl/dialects/riscv_scf.py:277]: @math-fehr what's the best way to test this?

Comment (AntonLydike) [reply xdsl/dialects/riscv_scf.py:277]: I guess a filecheck?

**Review by AntonLydike — APPROVED** [review]

File: xdsl/dialects/riscv_scf.py (hunk 1 above)
Comment (superlopuh) [reply xdsl/dialects/riscv_scf.py:277]: I tried it and it doesn't work, the whole thing fails as opposed to successfully finding the string with the error

Comment (superlopuh) [reply xdsl/dialects/riscv_scf.py:277]: `--parsing-diagnostics`

**Pull request merged**